modules/apps/commerce/commerce-payment-service/src/main/java/com/liferay/commerce/payment/internal/integration/FunctionCommercePaymentIntegration.java

```diff
@@ -234,13 +244,85 @@ private JSONObject _getPayloadJSONObject(
 		return JSONUtil.put(
 			"commercePaymentEntry",
 			_jsonFactory.looseSerializeDeep(commercePaymentEntry)
-		).put(
-			"objectEntry", _jsonFactory.looseSerializeDeep(objectEntry)
 		).put(
 			"typeSettings", typeSettingsJSONObject
 		);
 	}
 
+	private void _launchCatapult(
+			CommercePaymentEntry commercePaymentEntry, String resourcePath)
+		throws PortalException {
+
+		try {
+			TransactionCommitCallbackUtil.registerCallback(
+				() -> {
+					JSONObject jsonObject = _jsonFactory.createJSONObject(
+						new String(
+							_portalCatapult.launch(
+								commercePaymentEntry.getCompanyId(),
+								Http.Method.POST,
+								_functionCommercePaymentIntegrationConfiguration.
+									oAuth2ApplicationExternalReferenceCode(),
+								_getPayloadJSONObject(commercePaymentEntry),
+								resourcePath, commercePaymentEntry.getUserId()
+							).get()));
+
+					_toCommercePaymentEntry(commercePaymentEntry, jsonObject);
+
+					return null;
+				});
```

Comment:
@gamerson as you suggested I added a callback

Reply:
I don't believe this will have the effect you expect. Since this callback won't be executed until after the transaction commits you shouldn't see any changes to the commercePaymentEntry which you are subsequently passing to other catapult invocations, which means all the work with the response content is pointless, no?

In other words, all the methods like:
```java
	public CommercePaymentEntry cancel(
			HttpServletRequest httpServletRequest,
			CommercePaymentEntry commercePaymentEntry)
		throws PortalException {

		_launchCatapult(
			commercePaymentEntry,
			_functionCommercePaymentIntegrationConfiguration.cancelPath());

		return commercePaymentEntry;
	}
```
will never return a modified `commercePaymentEntry` because the transaction would need to be completed before those updates happen but you aren't waiting on it to complete. That being the case, you could reduce _launchCatapult to simply (not tested code):

```java
ExecutorService executor = ...

	private void _launchCatapult(
			CommercePaymentEntry commercePaymentEntry, String resourcePath)
		throws PortalException {

		try {
			executor.submit(
				() -> _portalCatapult.launch(
					commercePaymentEntry.getCompanyId(),
					Http.Method.POST,
					_functionCommercePaymentIntegrationConfiguration.
						oAuth2ApplicationExternalReferenceCode(),
					_getPayloadJSONObject(commercePaymentEntry),
					resourcePath, commercePaymentEntry.getUserId()
				)
			);
		}
		catch (Exception exception) {
			_log.error(exception);

			throw new PortalException(exception);
		}
	}
```